src/cargo/core/resolver/mod.rs

```diff
@@ -687,7 +721,7 @@ impl Context {
                                              feature)));
                 }
             }
-            ret.push((dep, base));
+            ret.push((Rc::new(dep.clone()), base));
```

Comment:
Looking at some profiles, this is actually pretty expensive. A `Dependency` is quite large and has a good deal of information inside it, so I think it may be best to actually just take the same strategy as `PackageId` here and have `Dependency` internally store an `Arc` with a `DependencyInner` that actually has all the data. That way you could just pass around `Dependency` by value here and it'd just be cheap to cloen.